### PR TITLE
kill off circular top-level directory dependencies

### PR DESCRIPTION
**Pull request by pcardune**
When it comes to the top-level directories in `apps/src`, we should really try to treat them like npm packages. I think everyone can agree that npm packages should not have circular dependencies between each other, and so we should set the same rule for ourselves among top-level directories.

This pull requests moves some files and code around so that this rule is followed.

Previously, we had the following cycles among top-level directories (including the root directory):

```
1) cycle from . → redux → . found in 1 different paths:
    StudioApp.js → 
    redux/pageConstants.js → 
    utils.js

2) cycle from . → templates → . found in 4 different paths:
    StudioApp.js → 
    templates/instructions/DialogInstructions.jsx → 
    StylelessRenderer.js

    StudioApp.js → 
    templates/instructions/DialogInstructions.jsx → 
    templates/instructions/Instructions.jsx → 
    experiments.js

    shareWarnings.js → 
    templates/ShareWarningsDialog.jsx → 
    color.js

    shareWarnings.js → 
    templates/ShareWarningsDialog.jsx → 
    templates/ShareWarnings.jsx → 
    color.js

3) cycle from applab → assetManagement → applab found in 1 different paths:
    applab/designMode.js → 
    assetManagement/assetPrefix.js → 
    applab/constants.js

4) cycle from applab → templates → applab found in 1 different paths:
    applab/AppLabCrosshairOverlay.jsx → 
    templates/CrosshairOverlay.jsx → 
    applab/gridUtils.js
```

and now these cycles are gone.

### CODE REVIEW

File: apps/src/redux/pageConstants.js [hunk 1]
```diff
@@ -1,9 +1,8 @@
 var _ = require('lodash');
-var utils = require('../utils');
```
Comment (davidsbailey) [inline apps/src/redux/pageConstants.js:2]: seems like `../utils` belongs under `../util/`

File: apps/src/util/color.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
+// /Users/pcardune/programming/code-dot-org/apps/src/util/color.js
+// GENERATED FILE: DO NOT MODIFY DIRECTLY
+// This generated file exports all variables defined in /Users/pcardune/programming/code-dot-org/shared/css/color.scss
+// for use in JavaScript. The generator script is convert-scss-variables.js
```
Comment (davidsbailey) [inline apps/src/util/color.js]: remove old generated `apps/src/color.js` ?

Comment (pcardune) [reply apps/src/util/color.js]: good call.

File: apps/src/redux/pageConstants.js (hunk 1 above)
Comment (pcardune) [reply apps/src/redux/pageConstants.js:2]: yeah...

**Comment by pcardune** [comment]
So, I have a script that does this kind of "structural" linting. But it is not quite ready to join our build pipeline. I'm going to continue running this script pretty frequently for my own work and will catch any regressions manually for the moment. But long term plan is to get the script into the build pipeline.

**Pull request merged**